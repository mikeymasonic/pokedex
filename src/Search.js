import React, { Component } from 'react'
export default class Search extends Component {
    state = {
        searchName: ''
    }
    handleForm = event => {
        if(!this.state.searchName){
            event.preventDefault();
            return;
        }
        event.preventDefault();
        const queryParam = new URLSearchParams();

        queryParam.set('pokemon', this.state.searchName);
        queryParam.set('page', 1);
        window.location.hash = queryParam.toString();
    }
    render() {
        return(
            <div id="search">
                <form onSubmit={this.handleForm}>
                <input id="search"
                onChange={e => this.setState({ searchName: e.target.value})} 
                value={this.state.searchName}/>
                <button id="submit">Search</button>
                </form>           
            </div>
        );
    }
}