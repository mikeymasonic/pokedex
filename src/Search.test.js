import React from 'react';
import Search from './Search.js';
import renderer from "react-test-renderer";

test('renders Search.js correctly', () => {
  const tree = renderer
  .create(<Search />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});