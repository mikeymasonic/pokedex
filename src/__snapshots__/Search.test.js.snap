// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Search.js correctly 1`] = `
<div
  id="search"
>
  <form
    onSubmit={[Function]}
  >
    <input
      id="search"
      onChange={[Function]}
      value=""
    />
    <button
      id="submit"
    >
      Search
    </button>
  </form>
</div>
`;
